File: app/src/main/java/azuwaredison89/gmail/dinacom/data/HargaDokterActivity.java
package azuwaredison89.gmail.dinacom.data;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;

import azuwaredison89.gmail.dinacom.R;
import azuwaredison89.gmail.dinacom.puskesmas.DaftarNamaDokterActivity;
import azuwaredison89.gmail.dinacom.puskesmas.DokterActivity;

public class HargaDokterActivity extends AppCompatActivity {

    protected ListView list_harga;
    protected ListAdapter adapter;
    SimpleAdapter Adapter;
    HashMap<String, String> mMap;
    ArrayList<HashMap<String, String>> list1;
    String[] nama1;
    String[] nama2;
    String[] nama3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_harga_dokter);

        list_harga = (ListView) findViewById(R.id.list_harga);

        nama1 = new String[] {"Hipertensi", "Arthritis (radang sendi)", "Stroke", "Kanker", "Batu ginjal"};
        nama2 = new String[] {"Rp. 200.000", "Rp. 200.000", "Rp. 200.000", "Rp. 200.000", "Rp. 200.000"};
        nama3 = new String[] {Integer.toString(R.drawable.obat), Integer.toString(R.drawable.obat),
                Integer.toString(R.drawable.obat),
                Integer.toString(R.drawable.obat),
                Integer.toString(R.drawable.obat) };

        list1 = new ArrayList<HashMap<String,String>>();

        for (int i = 0; i < nama1.length; i++){
            mMap = new HashMap<String, String>();
            mMap.put("list", nama1[i]);
            mMap.put("latin", nama2[i]);
            mMap.put("gbr", nama3[i]);
            list1.add(mMap);
        }

        Adapter = new SimpleAdapter(this, list1, R.layout.list_harga_dokter,
                new String[] {"list", "latin", "gbr"}, new int[] {R.id.tv_judul, R.id.tv_harga, R.id.iv_icon});
        list_harga.setAdapter(Adapter);

        list_harga.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            public void onItemClick(AdapterView<?> parent, View view,
                                    int position, long id) {
                String nama_negara=list_harga.getItemAtPosition(position).toString();
                Intent intent= new Intent(getApplicationContext(),
                        DataDokterActivity.class);
                startActivity(intent);
                finish();

            }
        });
    }
}

File: app/src/main/java/azuwaredison89/gmail/dinacom/ui/activity/pesandokter/HargaDokterActivity.kt
package azuwaredison89.gmail.dinacom.ui.activity.pesandokter

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListAdapter
import android.widget.ListView
import android.widget.SimpleAdapter
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import azuwaredison89.gmail.dinacom.R
import azuwaredison89.gmail.dinacom.model.DataDoctor
import azuwaredison89.gmail.dinacom.model.SpecificCategoryDoctor
import azuwaredison89.gmail.dinacom.ui.adapter.AdapterListDocter
import azuwaredison89.gmail.dinacom.ui.adapter.AdapterListHargaDocter
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import kotlinx.android.synthetic.main.activity_daftar_nama_dokter.*
import kotlinx.android.synthetic.main.activity_harga_dokter.*
import java.util.*

class HargaDokterActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_harga_dokter)

//        getListDoctorHarga()

    }

//    fun getListDoctorHarga() {
//        val idcategory = intent.getStringExtra("categoryDocter") as String
//        val idDoctor = intent.getStringExtra("idDocter") as String
//
//        val database = FirebaseDatabase.getInstance()
//        val ref = database.getReference("specific_category_doctor").child("79d8c39e-5a4b-11ea-8e2d-0242ac130003")
//        val listDocter = mutableListOf<SpecificCategoryDoctor>()
//        ref.addValueEventListener(object : ValueEventListener {
//            override fun onDataChange(p0: DataSnapshot) {
//                listDocter.clear()
//                for (snap in p0.children) {
//                    val user = snap.getValue(SpecificCategoryDoctor::class.java)
//                    user?.let {
//                    }
//                }
//
//                Log.d("apacok", listDocter[1].name)
//
//                val adapter = AdapterListHargaDocter(this@HargaDokterActivity, idDoctor, idcategory, listDocter as ArrayList<SpecificCategoryDoctor>)
//                recyclerViewPilihPenyakit.layoutManager = LinearLayoutManager(this@HargaDokterActivity, RecyclerView.VERTICAL, false)
//                recyclerViewPilihPenyakit.adapter = adapter
//                adapter.notifyDataSetChanged()
////                view.onSuccess(listMember)
//            }
//
//            override fun onCancelled(p0: DatabaseError) {
////                view.onFailed(p0.message)
//            }
//        })
    }
}